src/compiler/checker.ts

```diff
@@ -10671,11 +10671,13 @@ namespace ts {
             }
 
             if (!ignoreReturnTypes) {
-                const targetReturnType = getReturnTypeOfSignature(target);

```

Comment:
I definitely see your point about switching JS constructors to have a construct signature rather than a call signature. It's a major change at this point, though perhaps worth it.

In the meantime, I am tempted to propose `getEffectiveReturnType`, except that it doesn't help in signaturesRelatedTo.